src/utils.ts

```diff
@@ -10,3 +11,33 @@ export function onWorkspaceRootChange(cb: (rootPath: string) => void): void {
         }
     });
 }
+
+export function checkConfig(): PrettierVSCodeConfig {
+    const config: PrettierVSCodeConfig = workspace.getConfiguration(
+        'prettier'
+    ) as any;
+    if (config.useFlowParser) {
+        window.showWarningMessage(
+            "Option 'useFlowParser' has been deprecated. " +
+                'Use \'parser: "flow"\' instead.'
+        );
+    }
+    if (typeof config.trailingComma === 'boolean') {
+        window.showWarningMessage(
+            "Option 'trailingComma' as a boolean value has been deprecated. " +
+                "Use 'none', 'es5' or 'all' instead."
+        );
+    }
+    return config;
+}
+
+export function allEnabledLanguages(): DocumentSelector {
+    const config = checkConfig();
+    return [
+        ...config.javascriptEnable,
+        ...config.typescriptEnable,
+        ...config.cssEnable,
+        ...config.jsonEnable,
+        ...config.graphqlEnable,
+    ];
```

Comment:
That is very true, the reason I added it was to be a bit dynamic, if you change your settings you don't have to restart but actually you do, so maybe I don't have to compute that every single time.